kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/FeedbackInputFragment.java

```diff
@@ -0,0 +1,215 @@
+            throw new ClassCastException("Implement FeedbackFragmentListener in your parent fragment.");
+        }
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.feedback_rating_layout, container, false);
+
+        editTextFeedbackComment = view.findViewById(R.id.idEditTextFeedback);
+        buttonSubmitFeedback = view.findViewById(R.id.idButtonSubmit);
+
+        buttonsRating = new Button[3];
+        textViewsRatingText = new TextView[3];
+
+        buttonsRating[0] = view.findViewById(R.id.idButtonPoor);
+        buttonsRating[1] = view.findViewById(R.id.idButtonAverage);
+        buttonsRating[2] = view.findViewById(R.id.idButtonGood);
+        textViewsRatingText[0] = view.findViewById(R.id.idTextPoor);
+        textViewsRatingText[1] = view.findViewById(R.id.idTextAverage);
+        textViewsRatingText[2] = view.findViewById(R.id.idTextGood);
+        buttonCloseFragment = view.findViewById(R.id.idCloseFeedbackFragment);
+
+        editTextFeedbackComment.setVisibility(View.GONE);
+        textViewsRatingText[0].setVisibility(View.INVISIBLE);
+        textViewsRatingText[1].setVisibility(View.INVISIBLE);
+        textViewsRatingText[2].setVisibility(View.INVISIBLE);
+
+        buttonSubmitFeedback.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(ratingLevel == 0) {
+                    setRatingLevel(RATING_AVERAGE);
+                }
+                String feedbackComment = editTextFeedbackComment.getText().toString().trim();
+                feedbackFragmentListener.onFeedbackSubmitButtonPressed(getRatingLevel(), feedbackComment);
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        buttonCloseFragment.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        //set the un-selected size of the buttons to 80% of size
+        buttonsRating[0].setScaleX(0.8f);
+        buttonsRating[0].setScaleY(0.8f);
+        buttonsRating[1].setScaleX(0.8f);
+        buttonsRating[1].setScaleY(0.8f);
+        buttonsRating[2].setScaleX(0.8f);
+        buttonsRating[2].setScaleY(0.8f);
+
+        buttonsRating[0].setOnClickListener(this);
+        buttonsRating[1].setOnClickListener(this);
+        buttonsRating[2].setOnClickListener(this);
+
+        return view;
+    }
+
+    /**
+     * toggle button as selected or deselected
+     * @param select true or false
+     * @param b the button to toggle
+     */
+    void toggleRatingButtonSelected(boolean select, Button b) {
```

Comment:
@shubhamtewari please review all the code added and use proper naming everywhere, avoid names like a, b, c, d, 1, 2, 3, 4.